frontend/helpers/generateFakeSurplusAuction.ts

```diff
@@ -0,0 +1,74 @@
+import { SurplusAuction, SurplusAuctionData, SurplusAuctionStates, SurplusEvent } from 'auctions-core/src/types';
+import BigNumber from 'bignumber.js';
+import faker from 'faker';
+import { random } from 'lodash';
+import { generateFakeSurplusBidEvents, generateFakeSurplusEvent } from '~/helpers/generateFakeSurplusEvent';
+
+const SURPLUS_AUCTION_STATES: SurplusAuctionStates[] = [
+    'just-started',
+    'have-bids',
+    'ready-for-collection',
+    'collected',
+    'requires-restart',
+];
+
+export const generateFakeSurplusAuctionData = function (): SurplusAuctionData {
+    const state: SurplusAuctionStates = faker.helpers.randomize(SURPLUS_AUCTION_STATES);
+    const bidAmountMKR = new BigNumber(parseFloat(faker.finance.amount()));
+    const receiveAmountDAI = new BigNumber(parseFloat(faker.finance.amount()));
+    const receiverAddress = faker.finance.ethereumAddress();
+    const auctionEndDate = faker.date.soon();
+    const bidEndDate = state === 'ready-for-collection' ? faker.date.recent() : undefined;
+    const earliestEndDate = bidEndDate
+        ? auctionEndDate.getUTCMilliseconds() > bidEndDate.getUTCMilliseconds()
+            ? bidEndDate
+            : auctionEndDate
+        : auctionEndDate;
+
+    return {
+        bidAmountMKR,
+        receiveAmountDAI,
+        receiverAddress,
+        auctionEndDate,
+        bidEndDate,
+        earliestEndDate,
+        state,
+    };
+};
+
+export const generateFakeSurplusAuction = function (): SurplusAuction {
+    const surplusAuctionData = generateFakeSurplusAuctionData();
+    const id = faker.datatype.number();
+    const marketUnitPrice = new BigNumber(parseFloat(faker.finance.amount()));
+    const events: SurplusEvent[] = [];
+
+    events.push(generateFakeSurplusEvent('start'));
+
+    if (surplusAuctionData.state !== 'just-started') {
+        events.push(...generateFakeSurplusBidEvents());
+    }
+
+    if (surplusAuctionData.state === 'collected') {
+        events.push(generateFakeSurplusEvent('collect'));
+    }
+
+    const latestBids = events.filter(event => {
+        return event.type === 'bid';
+    });
+    const highestBid = latestBids.length > 0 ? latestBids[latestBids.length - 1].bidAmountMKR : undefined;
```

Comment:
I currently calculate the highest bid and auction price in the helper function. Meaning we will later need to recalculate this every X seconds, like with the step/cut price of normal auctions. 

Do we want to do this, or should I have this number be calculated inside the component. My thought process of calculating it externally was, that we want to show this data in other parts of our UI as well, meaning I personally belive it makes more sense to calculate it once, globally rather then multiple times in seperate components. However, this could also be changed if we belive it is better to do it on a component level.

Reply:
I'm on the same side here. I think calculating inside components would just lead to duplicate code, and I don't see any upsides compared to calculating externally.

Reply:
`bidAmountMKR` is actually the field that always contains the highest bid

Also, this PR should remove events as we had to change implementation proposal based on investigation.

Reply:
Do we have any central place, where we defined the types? I did some looking and I saw that you updated the types in this issue: https://github.com/sidestream-tech/unified-auctions-ui/issues/188#issuecomment-1161592228

However, the types in the PR #317 are slightly different. Which one should I use?

Reply:
Feel free to use any of those, since ones in the #317 might not be final as well

Reply:
What is missing in the types in #188 is the address of the highest bidder. We want to display in the UI if the current user is the highest bidder and to make this possible we need to store the address.

Is this technological possible?